In [1]:
# Loading the required libraries

import pandas as pd
import numpy as np

In [2]:
# Reding the file
df=pd.read_csv('diabetes.csv')

In [3]:
# Viewing the data in file
df.head()

Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
0            6      148             72             35        0  33.6   
1            1       85             66             29        0  26.6   
2            8      183             64              0        0  23.3   
3            1       89             66             23       94  28.1   
4            0      137             40             35      168  43.1   

   DiabetesPedigreeFunction  Age  Outcome  
0                     0.627   50        1  
1                     0.351   31        0  
2                     0.672   32        1  
3                     0.167   21        0  
4                     2.288   33        1

In [4]:
# checking how many rows & columns are present in dataset
df.shape

(768, 9)

In [5]:
# Statistical values of the data

df.describe()

Pregnancies     Glucose  BloodPressure  SkinThickness     Insulin  \
count   768.000000  768.000000     768.000000     768.000000  768.000000   
mean      3.845052  120.894531      69.105469      20.536458   79.799479   
std       3.369578   31.972618      19.355807      15.952218  115.244002   
min       0.000000    0.000000       0.000000       0.000000    0.000000   
25%       1.000000   99.000000      62.000000       0.000000    0.000000   
50%       3.000000  117.000000      72.000000      23.000000   30.500000   
75%       6.000000  140.250000      80.000000      32.000000  127.250000   
max      17.000000  199.000000     122.000000      99.000000  846.000000   

              BMI  DiabetesPedigreeFunction         Age     Outcome  
count  768.000000                768.000000  768.000000  768.000000  
mean    31.992578                  0.471876   33.240885    0.348958  
std      7.884160                  0.331329   11.760232    0.476951  
min      0.000000                  0.078000   21.000000    0.000000  
25%     27.300000                  0.243750   24.000000    0.000000  
50%     32.000000                  0.372500   29.000000    0.000000  
75%     36.600000                  0.626250   41.000000    1.000000  
max     67.100000                  2.420000   81.000000    1.000000

In [6]:
# Getting the count of Non-Diabetics & Diabetics
# 0: Non-Diabetic & 1:Diabetic

df['Outcome'].value_counts()

0    500
1    268
Name: Outcome, dtype: int64

Observation:- Out of 768 people 500 people are Non-diabetic & 268 are Diabetic.

In [7]:
# Finding the Average levels(mean) of measures(features) for Diabetic & Non-Diabetic

df.groupby('Outcome').mean()

Pregnancies     Glucose  BloodPressure  SkinThickness     Insulin  \
Outcome                                                                      
0           3.298000  109.980000      68.184000      19.664000   68.792000   
1           4.865672  141.257463      70.824627      22.164179  100.335821   

               BMI  DiabetesPedigreeFunction        Age  
Outcome                                                  
0        30.304200                  0.429734  31.190000  
1        35.142537                  0.550500  37.067164

In [8]:
# Separating the features or columns
X= df.drop(columns= 'Outcome', axis=1)
Y= df['Outcome']

In [9]:
X

Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
0              6      148             72             35        0  33.6   
1              1       85             66             29        0  26.6   
2              8      183             64              0        0  23.3   
3              1       89             66             23       94  28.1   
4              0      137             40             35      168  43.1   
..           ...      ...            ...            ...      ...   ...   
763           10      101             76             48      180  32.9   
764            2      122             70             27        0  36.8   
765            5      121             72             23      112  26.2   
766            1      126             60              0        0  30.1   
767            1       93             70             31        0  30.4   

     DiabetesPedigreeFunction  Age  
0                       0.627   50  
1                       0.351   31  
2                       0.672   32  
3                       0.167   21  
4                       2.288   33  
..                        ...  ...  
763                     0.171   63  
764                     0.340   27  
765                     0.245   30  
766                     0.349   47  
767                     0.315   23  

[768 rows x 8 columns]

In [10]:
Y

0      1
1      0
2      1
3      0
4      1
      ..
763    0
764    0
765    0
766    1
767    0
Name: Outcome, Length: 768, dtype: int64

# Data Standardization

In [11]:
from sklearn.preprocessing import StandardScaler
scaler = StandardScaler()

In [12]:
x=scaler.fit_transform(X)

In [13]:
x

array([[ 0.63994726,  0.84832379,  0.14964075, ...,  0.20401277,
         0.46849198,  1.4259954 ],
       [-0.84488505, -1.12339636, -0.16054575, ..., -0.68442195,
        -0.36506078, -0.19067191],
       [ 1.23388019,  1.94372388, -0.26394125, ..., -1.10325546,
         0.60439732, -0.10558415],
       ...,
       [ 0.3429808 ,  0.00330087,  0.14964075, ..., -0.73518964,
        -0.68519336, -0.27575966],
       [-0.84488505,  0.1597866 , -0.47073225, ..., -0.24020459,
        -0.37110101,  1.17073215],
       [-0.84488505, -0.8730192 ,  0.04624525, ..., -0.20212881,
        -0.47378505, -0.87137393]])

# TRAIN & TEST SPLIT the data

In [14]:
from sklearn.model_selection import train_test_split

X_train, X_test, Y_train, Y_test= train_test_split(x,Y,test_size=0.2,stratify=Y, random_state=2)

In [15]:
x.shape , X_train.shape, X_test.shape

((768, 8), (614, 8), (154, 8))

# Training the Model

In [17]:
from sklearn import svm
classifier =svm.SVC(kernel='linear')

In [18]:
#Training the Support Vector Machine Classifier
classifier.fit(X_train, Y_train)

SVC(kernel='linear')

Accuracy score checking

In [19]:
# Accuracy score with the training data

from sklearn.metrics import accuracy_score

X_train_prediction=classifier.predict(X_train)
tr_accuracy= accuracy_score(X_train_prediction,Y_train)

In [20]:
print('Accuracy of the training data: ',tr_accuracy)

Accuracy of the training data:  0.7866449511400652


In [21]:
# Accuracy score with the testing data

X_test_prediction=classifier.predict(X_test)
ts_accuracy= accuracy_score(X_test_prediction,Y_test)

In [22]:
print('Accuracy of the testing data: ',ts_accuracy)

Accuracy of the testing data:  0.7727272727272727


# checking the system by giving INPUT

In [26]:
input_data=(8,99,84,0,0,35.4,0.388,50)
#converting the input data to numpy array
input_data_as_numpy_array= np.asarray(input_data)

#reshaping the array as we are predicting for one instance
input_data_reshaped=input_data_as_numpy_array.reshape(1,-1)

#standardizing the input data
std_data=scaler.transform(input_data_reshaped)
print(std_data)

prediction=classifier.predict(std_data)
print(prediction)

if prediction[0]==0:
    print('The person is Not-Diabetic')
else:
    print('The person is Diabetic')

[[ 1.23388019 -0.68523633  0.77001375 -1.28821221 -0.69289057  0.43246741
  -0.25331639  1.4259954 ]]
[0]
The person is Not-Diabetic


C:\Users\ajays\anaconda3\lib\site-packages\sklearn\base.py:450: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names
  warnings.warn(


In [27]:
input_data=(4,70,90,25,0,34,0.388,25)
#converting the input data to numpy array
input_data_as_numpy_array= np.asarray(input_data)

#reshaping the array as we are predicting for one instance
input_data_reshaped=input_data_as_numpy_array.reshape(1,-1)

#standardizing the input data
std_data=scaler.transform(input_data_reshaped)
print(std_data)

prediction=classifier.predict(std_data)
print(prediction)

if prediction[0]==0:
    print('The person is Not-Diabetic')
else:
    print('The person is Diabetic')

[[ 0.04601433 -1.59285354  1.08020025  0.27998931 -0.69289057  0.25478047
  -0.25331639 -0.70119842]]
[0]
The person is Not-Diabetic


C:\Users\ajays\anaconda3\lib\site-packages\sklearn\base.py:450: UserWarning: X does not have valid feature names, but StandardScaler was fitted with feature names
  warnings.warn(
